src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -199,6 +201,55 @@ case class MatrixValue(
     new UnpartitionedRVD(signature, hc.sc.parallelize(colValues.value.toArray.map(_.asInstanceOf[Row]))
       .mapPartitions(_.toRegionValueIterator(signature)))
   }
+
+  def entriesRVD(): RVD = {
+    val resultStruct = typ.entriesTableType.rowType
+    val fullRowType = typ.rvRowType
+    val localEntriesIndex = typ.entriesIdx
+    val localEntriesType = typ.entryArrayType
+    val localColType = typ.colType
+    val localEntryType = typ.entryType
+    val localNCols = nCols
+
+    val localColValuesBc = colValues.broadcast
+
+    rvd.mapPartitions(resultStruct) { it =>
+
+      val colValues = localColValuesBc.value
```

Comment:
probably worth converting this to a RV once, because it'll be referenced a huge number of times.